crates/polars-plan/src/dsl/function_expr/pow.rs

```diff
@@ -54,38 +91,79 @@ where
             _ => base.apply(|v| Pow::pow(v, exponent_value)).into_series(),
         };
         Ok(Some(s))
-    } else if (base.len() == 1) && (exponent.len() != 1) {
-        let base = base
-            .get(0)
-            .ok_or_else(|| polars_err!(ComputeError: "base is null"))?;
+    } else {
+        pow_on_chunked_arrays(base, exponent)
+    }
+}
 
-        Ok(Some(
-            exponent.apply(|exp| Pow::pow(base, exp)).into_series(),
-        ))
+fn pow_on_integral_dtypes<T, F>(
+    base: &ChunkedArray<T>,
+    exponent: &ChunkedArray<F>,
+) -> PolarsResult<Option<Series>>
+where
+    T: PolarsIntegerType,
+    F: PolarsIntegerType,
+    T::Native: num::pow::Pow<F::Native, Output = T::Native> + ToPrimitive,
+    ChunkedArray<T>: IntoSeries,
+{
+    let dtype = T::get_dtype();
+
+    if exponent.len() == 1 {
+        let Some(exponent_value) = exponent.get(0) else {
+            return Ok(Some(Series::full_null(base.name(), base.len(), &dtype)));
+        };
+        let s = match exponent_value.to_u64().unwrap() {
+            a if a == 1 => base.clone().into_series(),
+            a if a < 10 && a > 1 => {
+                let mut out = base.clone();
+
+                for _ in 1..exponent_value.to_u8().unwrap() {
+                    out = out * base.clone()
+                }
+                out.into_series()
+            },
+            _ => base.apply(|v| Pow::pow(v, exponent_value)).into_series(),
+        };
+        Ok(Some(s))
     } else {
-        Ok(Some(
-            base.into_iter()
-                .zip(exponent)
-                .map(|(opt_base, opt_exponent)| match (opt_base, opt_exponent) {
-                    (Some(base), Some(exponent)) => Some(num::pow::Pow::pow(base, exponent)),
-                    _ => None,
-                })
-                .collect_trusted::<ChunkedArray<T>>()
-                .into_series(),
-        ))
+        pow_on_chunked_arrays(base, exponent)
     }
 }
 
 fn pow_on_series(base: &Series, exponent: &Series) -> PolarsResult<Option<Series>> {
     use DataType::*;
-    match base.dtype() {
-        Float32 => {
+    match (base.dtype(), exponent.dtype()) {
+        (UInt8 | UInt16 | UInt32, UInt8 | UInt16 | UInt32) => {
+            let base = base.strict_cast(&DataType::UInt32)?;
+            let ca = base.u32().unwrap();
+            let exponent = exponent.strict_cast(&DataType::UInt32)?;
+            pow_on_integral_dtypes(ca, exponent.u32().unwrap())
+        },
```

Comment:
I'm casting these to UInt32 as, trying to pass a ChunkedArray<UInt16> to `pow_on_integral_dtypes` gives
```rust
the trait bound `implementations::SeriesWrap<polars_core::chunked_array::ChunkedArray<UInt16Type>>: polars_core::prelude::SeriesTrait` is not satisfied
the following other types implement trait `polars_core::prelude::SeriesTrait`:
  implementations::SeriesWrap<Logical<DateType, polars_core::datatypes::Int32Type>>
  implementations::SeriesWrap<Logical<DatetimeType, polars_core::datatypes::Int64Type>>
  implementations::SeriesWrap<Logical<DurationType, polars_core::datatypes::Int64Type>>
...
```

Reply:
I don't understand? Isn't that because you would send a series otherwise?

Reply:
sorry I mean that I get that error if I do `exponent.u16()` instead of `exponent.u32()`

Reply:
I was just missing `#[cfg(feature = "dtype-u16")]`, solved now